### ISSUE TITLE
Enable multi config with CLI

### ISSUE THREAD

**Issue by georeith** [webpack/webpack#2835]
I'm submitting a feature request that multiple `--config` flags can be passed to the CLI to enable multi-compiler mode. For example:
```
webpack --config webpack/config.lib.js --config webpack/config.web.js
```

**Webpack version:**
1.13.x

**Please tell us about your environment:**
OSX 10.x

**Current behavior:**
```
TypeError: Path must be a string. Received [ 'webpack/config.lib.js', 'webpack/config.web.js' ]
```

**Expected/desired behavior:**

* **What is the expected behavior?**
  
  Webpack treats it the same as receiving a path to a config file that returns an array of configurations. Possibly flattening any arrays it receives (not strictly necessary just a useful feature).

* **What is the motivation / use case for changing the behavior?**

  It allows you to keep your configurations separate yet build them simultaneously and efficiently using the CLI

**Comment by asarode** [comment]
I think this would be useful! I'm down to work on it.

This is my first time working on this codebase so any suggestions on the implementation would be helpful. I'll dive in soon and flesh out any questions I have.

**Comment by SpaceK33z** [comment]
This can be closed, right? #3007 was merged and part of release `v2.1.0-beta.23`.
